evm_loader/test_delete_account.py

```diff
@@ -0,0 +1,150 @@
+        return TransactionInstruction(program_id=self.loader.loader_id, data=bytearray.fromhex("05") + trx_data, keys=[
+                    AccountMeta(pubkey=owner_contract, is_signer=False, is_writable=True),
+                    AccountMeta(pubkey=contract_code, is_signer=False, is_writable=True),
+                    AccountMeta(pubkey=self.caller, is_signer=False, is_writable=True),
+                    AccountMeta(pubkey=PublicKey("Sysvar1nstructions1111111111111111111111111"), is_signer=False, is_writable=False),
+                    AccountMeta(pubkey=self.loader.loader_id, is_signer=False, is_writable=False),
+                    AccountMeta(pubkey=PublicKey("SysvarC1ock11111111111111111111111111111111"), is_signer=False, is_writable=False),
+                ])
+
+
+    def test_success_tx_send(self):
+        (owner_contract, contract_code) = self.deploy_contract()
+
+        init_nonce = getTransactionCount(client, self.caller)
+        (trx_data_1, keccak_instruction_1) = self.make_transaction(owner_contract, init_nonce, 1)
+        init_nonce += 1
+        (trx_data_2, keccak_instruction_2) = self.make_transaction(owner_contract, init_nonce, 3)
+
+        trx = Transaction().add(
+                self.sol_instr_keccak(keccak_instruction_1)
+            ).add(
+                self.sol_instr_call(trx_data_1, owner_contract, contract_code)
+            ).add(
+                self.sol_instr_keccak(keccak_instruction_2)
+            ).add(
+                self.sol_instr_call(trx_data_2, owner_contract, contract_code)
+            )
+
+        err = "invalid account data for instruction"
+        with self.assertRaisesRegex(Exception,err):
+            result = send_transaction(client, trx, self.acc)
+            print(result)
+
+
+    def test_funds_transfer(self):
+        (owner_contract, contract_code) = self.deploy_contract()
+
+        init_nonce = getTransactionCount(client, self.caller)
+        (trx_data_1, keccak_instruction_1) = self.make_transaction(owner_contract, init_nonce, 1)
+
+        caller_balance_pre = getBalance(self.caller)
+        contract_balance_pre = getBalance(owner_contract)
+
+        trx = Transaction().add(
+                self.sol_instr_keccak(keccak_instruction_1)
+            ).add(
+                self.sol_instr_call(trx_data_1, owner_contract, contract_code)
+            )
+
+        send_transaction(client, trx, self.acc)
+
+        caller_balance_post = getBalance(self.caller)
+        contract_balance_post = getBalance(owner_contract)
+
+        self.assertEqual(caller_balance_post, contract_balance_pre + caller_balance_pre)
+        self.assertEqual(contract_balance_post, 0)
+
+
+    def test_success_deletion(self):
+        (owner_contract, contract_code) = self.deploy_contract()
+
+        init_nonce = getTransactionCount(client, self.caller)
+        (trx_data_1, keccak_instruction_1) = self.make_transaction(owner_contract, init_nonce, 1)
+
+        trx = Transaction().add(
```

Comment:
You should check balance changes and that you can create a Solana account which will be destroyed.